lib/embulk/output/bigquery/bigquery_client.rb

```
@@ -411,7 +411,7 @@ def create_table_if_not_exists(table, dataset: nil, options: nil)
             dataset ||= @dataset
             options ||= {}
             options['time_partitioning'] ||= @task['time_partitioning']
-            if Helper.has_partition_decorator?(table)
+            if Helper.has_partition_decorator?(table) && @task['range_partitioning'].nil?
```

I think it would be better to throw an error if a table contains $ and range_partitioning is set.

```yaml
out:
  type: bigquery
  mode: append
  project: project
  dataset: dataset
  auth_method: service_account
  json_keyfile: key.json
  auto_create_table: true
  table: part_test$hoge
  location: asia-northeast1
  range_partitioning:
    field: id
    range:
      start: 1
      end: 5001
      interval: 500
```

Probably current implementation try to create normal table and raise the exception.

```
Error: org.jruby.exceptions.StandardError: (Error) failed to create table project:dataset.LOAD_TEMP_d553f425_6740_488f_9522_a242c164fba3_part_test$hoge in asia-northeast1, \
response:{:status_code=>400, :message=>"invalid: Invalid table ID \"LOAD_TEMP_d553f425_6740_488f_9522_a242c164fba3_part_test$hoge\". \
Table IDs must be alphanumeric (plus underscores) and must be at most 1024 characters long. Also, Table decorators cannot be used.", :error_class=>Google::Apis::ClientError}
```